### PR TITLE
[api] fix errors unhandled

### PR DESCRIPTION
**Pull request by huof6829**
# Description
fix api#96~101  [Errors unhandled.](https://github.com/iotexproject/iotex-core/security/code-scanning)  
fix  blockchain#95  [Errors unhandled.](https://github.com/iotexproject/iotex-core/security/code-scanning/95)  
fix db#94 [Errors unhandled.](https://github.com/iotexproject/iotex-core/security/code-scanning/94)

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/listener.go [hunk 1]
```diff
@@ -136,8 +140,10 @@ func newIDGenerator(length uint8) *randID {
 	return &randID{length: length}
 }
 
-func (id *randID) newID() string {
+func (id *randID) newID() (string, error) {
 	token := make([]byte, id.length)
-	rand.Read(token)
-	return "0x" + hex.EncodeToString(token)
+	if _, err := rand.Read(token); err != nil {
```
Comment (Liuhaai) [inline api/listener.go:145]: Read generates len(p) random bytes from the default Source and writes them into p. It always returns len(p) and a nil error.  Read, unlike the Rand.Read method, is safe for concurrent use.

Comment (huof6829) [reply api/listener.go]: ok, revert

File: api/websocket.go [hunk 2]
```diff
@@ -58,10 +58,14 @@ func (wsSvr *WebsocketHandler) ServeHTTP(w http.ResponseWriter, req *http.Reques
 
 func (wsSvr *WebsocketHandler) handleConnection(ws *websocket.Conn) {
 	defer ws.Close()
-	ws.SetReadDeadline(time.Now().Add(pongWait))
+	if err := ws.SetReadDeadline(time.Now().Add(pongWait)); err != nil {
```
Comment (Liuhaai) [inline api/websocket.go:61]: failed to set readdeadline

File: api/websocket.go [hunk 3]
```diff
@@ -83,7 +87,9 @@ func (wsSvr *WebsocketHandler) handleConnection(ws *websocket.Conn) {
 			err = wsSvr.msgHandler.HandlePOSTReq(reader,
 				apitypes.NewResponseWriter(
 					func(resp interface{}) error {
-						ws.SetWriteDeadline(time.Now().Add(writeWait))
+						if err = ws.SetWriteDeadline(time.Now().Add(writeWait)); err != nil {
+							log.Logger("api").Warn("failed to set write deadline timeout.", zap.Error(err))
```
Comment (Liuhaai) [inline api/websocket.go:91]: same here

File: api/websocket.go [hunk 4]
```diff
@@ -100,15 +106,19 @@ func ping(ctx context.Context, ws *websocket.Conn, cancel context.CancelFunc) {
 	pingTicker := time.NewTicker(pingPeriod)
 	defer func() {
 		pingTicker.Stop()
-		ws.Close()
+		if err := ws.Close(); err != nil {
```
Comment (Liuhaai) [inline api/websocket.go:109]: the connection is dead. The logging is not needed here
_ = ws.close()

**Comment by Liuhaai** [comment]
others look good for me

**Review by Liuhaai — APPROVED** [review]

File: api/websocket.go (hunk 2 above)
Comment (huof6829) [reply api/websocket.go:61]: no need to log if err!=nil ?

**Review by dustinxie — APPROVED** [review]

File: api/websocket.go (hunk 2 above)
Comment (Liuhaai) [reply api/websocket.go:61]: sorry comment in the wrong line

**Pull request merged**